baqend/top100CRUX.ts

```diff
@@ -0,0 +1,242 @@
+  'https://www.fressnapf.de',
+  'https://www.buecher.de'
+];
+
+/**
+ * Extracts CrUX report results for the given list of urls.
+ *
+ * @param db
+ * @param month
+ * @param year
+ * @param device
+ * @returns List of results as tab-formatted string.
+ */
+async function getFormattedResults(db: baqend, month: number, year: number, device: string) {
+  let formattedResult = '';
+
+  for (const url of TOP_100) {
+    const report = await db.ChromeUXReport.find()
+      .equal('url', url)
+      .equal('month', month)
+      .equal('year', year)
+      .equal('device', device)
+      .singleResult();
+    if (report.status === 'SUCCESS') {
+      formattedResult += `${report.url}\t`;
+      formattedResult += `${report.fcpMedian}\t`;
+      formattedResult += `${report.fpMedian}\t`;
+      formattedResult += `${report.dclMedian}\t`;
+      formattedResult += `${report.olMedian}\n`;
+    }
+  };
+
+  return formattedResult;
+}
+
+/**
+ * Checks if CrUX report for the given parameters is processed completely.
+ *
+ * @param db
+ * @param params
+ */
+async function checkStatus(db: baqend, params: QueriedParams) {
+  const reports = await db.ChromeUXReport.find()
+    .equal('url', params.url)
+    .equal('month', params.month)
+    .equal('year', params.year)
+    .resultList();
+  for(const report of reports) {
+    if (report.status && report.status === 'RUNNING') {
+      return false;
+    }
+  }
+  return true;
+}
+
+/**
+ * Starts fetching of CrUX reports for the given list of urls.
+ *
+ * @param db
+ * @param month
+ * @param year
+ */
+async function fetchCruxReports(db: baqend, month: number, year: number): Promise<void> {
+  for(const url of TOP_100) {
+    const params: QueriedParams = { url, month, year };
+    const responseMessage = await fetchCruxReport(db, params);
+    if (responseMessage === ResponseMessage.Success) {
+      await cruxFinished(db, params);
+      db.clear();
+    }
+  }
+}
+
+/**
+ * Periodically checks whether a specified CrUX report ist finished.
+ *
+ * @param db
+ * @param params
+ */
+function cruxFinished(db: baqend, params: QueriedParams): Promise<boolean> {
+  return new Promise((resolve) => {
+    let retry = 0;
+    const interval = setInterval(async () => {
+      const done = await checkStatus(db, params);
+      if(done){
+        clearInterval(interval);
+        resolve(done);
+      }
+      if(retry >= 6) {
+        clearInterval(interval);
+        resolve(false);
+      }
+      retry++;
+    },10000);
+  });
+}
+
+/**
+ * post interface to start fetching Chrome User Experience Reports for top 100.
+ *
+ * @param db
+ * @param request
+ * @param response
+ */
+export async function post(db: baqend, request: Request, response: Response) {
+  const { month, year } = request.body;
+  fetchCruxReports(db, month, year);
+
+  response.send( {message: 'Started fetching ChromeUX reports.'});
+}
+
+/**
+ * get interface to extract Chrome User Experience Report results for top 100.
```

Comment:
Start comment with capital, add line between description and parameters and add comments to all parameters